rrmngmnt/firewall.py

```diff
@@ -0,0 +1,143 @@
+from rrmngmnt.service import Service
+
+IPTABLES = 'iptables'
+
+
+class Firewall(Service):
+    """
+    Class for firewall services
+    """
+    def __init__(self, host, firewall_service):
+        super(Firewall, self).__init__(host)
+        self.host = host
+        self.firewall_service = firewall_service
+
+    def is_active(self):
+        return self.host.service(self.firewall_service).status()
+
+    def config_firewall(
+        self, action, chain, address_type, dest, target, protocol='all',
+        ports=None
+    ):
+        """
+        Changes firewall configuration
+
+        Args:
+           action (str): action to perform
+           chain (str): affected chain name
+           address_type (str): '--destination' for outgoing rules,
+               '--source' for incoming
+           dest (dict): 'address' key and value containing destination host or
+               list of destination hosts
+           target (str): target rule to apply
+           protocol (str): affected network protocol, Default is 'all'
+           ports (list): list of ports to configure
+
+       Returns:
+           bool: True if configuration change succeeded, False otherwise
+
+       Raises:
+           NotImplementedError: In case the users specifies more than 15 ports
+                to block
+
+       Example:
+           config_firewall(service='iptables', action='--append',
+                chain='OUTPUT', dest={'address': nfs_server}, target='DROP'
+        """
+        assert self.is_active(), (
+            "Service %s is not running on host" % self.firewall_service
+        )
+
+        dest = ",".join(dest['address'])
+        cmd = [
+            self.firewall_service, action, chain, address_type, dest, '--jump',
+            target.upper(), '--protocol', protocol
+        ]
+
+        if ports:
+            # Iptables multiport module accepts up to 15 ports
+            if len(ports) > 15:
+                raise NotImplementedError("Up to 15 ports can be specified")
+            ports = ",".join(ports)
+
+            if protocol.lower() == 'all':
+                # Adjust the protocol type, '--dports' option requires specific
+                # type
+                cmd.extend([protocol, 'tcp'])
+
+            cmd.extend(['--match', 'multiport', '--dports', ports])
+
+        return not self.host.executor().run_cmd(cmd)[0]
+
+
+class Chain(Firewall):
```

Comment:
I didn't mean to be chain subcalss of firewall.

I meant it as structure used by firewall.

```
h = Host("x.y.z")
h.firewall.chain("OUTPUT").add_rule(...)
```